clusterloader2/clean-up-old-snapshots.sh

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+print-usage-and-exit() {
+  >&2 echo "Usage:"
+  >&2 echo "    $0 delete|list <comma-sep-projects> <comma-sep-prefixes> <days-old>"
+  >&2 echo "Example:"
+  >&2 echo "    $0 list k8s-e2e-gce-1-1,k8s-e2e-gce-1-2  ci-e2e-scalability,ci-e2e-kubemark 30"
+  exit $1
+}
+
+if [ "$#" -ne 4 ]; then
+  >&2 echo "Wrong number of parameters, expected 4, got $#"
+  print-usage-and-exit 1
+fi
+
+command=$1
+projects=$2
+prefixes=$3
+days_old=$4
+
+if ! [[ "$projects" =~ ^[a-z0-9,-]+$ ]]; then
+  >&2 echo "Illegal characters in projects parameter: $projects"
+  print-usage-and-exit 2
+fi
+if ! [[ "$prefixes" =~ ^[a-z0-9,.-]+$ ]]; then
+  >&2 echo "Illegal characters in prefixes parameter: $prefixes"
+  print-usage-and-exit 3
+fi
+if ! [[ "$days_old" =~ ^[0-9]+$ ]]; then
+  >&2 echo "Days-old parameter must be an integer: $days_old"
+  print-usage-and-exit 4
+fi
+
+list-old-snapshots() {
+  project=$1
+  regexp=$2
+  days_old=$3
+  (
+    set -o xtrace
+    gcloud compute snapshots list --project "$project" --filter="creationTimestamp < -P${days_old}D AND name~\"${regexp}\"" --format="value(name)"
```

Comment:
nit: move each flag to a separate line - it makes it easier to read and track changes over the time

Reply:
done